### PR TITLE
updating the group ids and adding groups to drafts

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
## MADiE PR

Jira Ticket: [MAT-5237](https://jira.cms.gov/browse/MAT-5237)
(Optional) Related Tickets:

### Summary
updating the group ids and adding groups to drafts

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/VersionService.java [hunk 1]
```diff
@@ -114,6 +116,14 @@ public Measure createDraft(String id, String measureName, String username) {
     return savedDraft;
   }
 
+  private List<Group> updateGroupIds(List<Group> groups) {
+    if (groups != null) {
+      groups.stream().forEach(group -> group.setId(ObjectId.get().toString()));
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/VersionService.java]: I don't think you would want to update the original measure groups. Please create copies.

Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/services/VersionService.java]: I have done in similar manner earlier but later changed it, and forgot to revert the changes. Thanks for noticing I'll fix it

File: src/main/java/cms/gov/madie/measure/services/VersionService.java [hunk 2]
```diff
@@ -114,6 +116,15 @@ public Measure createDraft(String id, String measureName, String username) {
     return savedDraft;
   }
 
+  private List<Group> updateGroupIds(List<Group> groups) {
+    if (groups != null) {
+      List<Group> updatedGroups = groups;
+      updatedGroups.forEach(group -> group.setId(ObjectId.get().toString()));
+      return updatedGroups;
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/VersionService.java]: You are creating a new reference to an existing variable. you are still modifying existing objects. you can try this

```
return groups.stream()
          .map(group -> group.toBuilder().id(ObjectId.get().toString()).build())
          .collect(Collectors.toList());
```
Also, I think the method name is not so appropriate to the task it is doing. Probably `cloneMeasureGroups(List<Group> groups)  ` or something similar

Comment (sb-prateekkeerthi) [reply src/main/java/cms/gov/madie/measure/services/VersionService.java]: Thank you, I have updated the code

**Review by adongare — APPROVED** [review]

**Pull request merged**